src/FileHandler.cpp

```diff
@@ -76,12 +76,15 @@ std::string FileHandler::getFileResource(HttpRequest *request, std::ios_base::op
 	std::string fullPath = getFilePath(request->getResourcePath());
 
 	struct stat file_status;
-	if ((stat(fullPath.c_str(), &file_status) == 0) && S_ISDIR(file_status.st_mode))
+	if (request->getMethod() != Util::METHOD::POST)
```

Comment:
wait, why this check here? isnt this just getting our own resource to read, not to worry about requests here.

Reply:
maybe this whole function should just take in the resourcepath, not the whole request..

Reply:
I added it in, cause when doing the post our "resource path" is just the folder we want to upload stuff to. So when parsing the response we would trigger this, and instead of writing out a successful post request and fetching 201 we actually throw an error and write an error response. does this make sense to you.

Reply:
![image](https://github.com/rosamakinen/webserv/assets/44494010/4f27dacc-5b1f-4cf4-9c50-47790353bcc7)
here is where you should decide the file resource thingie, thats why it was adding the error response!

Reply:
can you explain this further? I guess the reason my implementation was that, since we have a system to fetch the files from the error pages -folder, do all the checks for it etc, I wanted to use that system. thats why I said we could change the name from error_pages to response_pages. 

Also I think my brain is fried, i'm still at school, so I also don't just cannot understand what you mean by "file resource thingie" and what "it" means that. My implementation was not probably the best, but my idea for it was to use the system we already had built in to do the checks and output the file that is in the same folder as the error pages are.

Reply:
Yeah so the idea for this function is to fetch the correct file based on the path given to it. In the filehandler we should not really even give the request as it makes no sense at the point of writing a response to consider the request at all! this method is supposed to be very general and only used to get a path and read a file based on thay path. so tve directory lisring etc needs to be taken off.

I believe that you get the error here because you are calling this function in response->setResponse and as the status is 201 and not exactly 200, the setResponse guides to to the error page without this bypass you did here. But this is not the pkace to do the directory listing and / or bypass error responses by checking the requested method! Rather create abother function for your special 201 and 204 files if they need this special thing than possibly bypass all possible outgoing responses with this check here! I can move the directory listing too so you can move this bypass.

Reply:
aka Im afraid this check will mess up other responses too and thus check really belongs somewhere else than in the filehandler  Filehandler should bring out files base on the response status, not the request method.

Reply:
thus said I think you tmhave done ylur best, but just this check needs ti be moved as well as the request as a parameter and directory listing from this function